XenAdmin/Wizards/ImportWizard/ImportSelectNetworkPage.cs

```diff
@@ -51,6 +54,31 @@ protected override bool ImplementsIsDirty()
             return true;
         }
 
+        protected override bool ShowNetwork(Host targetHost, XenAPI.Network network, string sysId = null)
+        {
+            if (network.IsSriov() && !AllowSriovNetwork(sysId))
+                return false;
+
+            return base.ShowNetwork(targetHost, network);

```

Comment:
It's not a very good idea to call the base class method without the optional parameter. If someone changes the default value in the overriding method (not a good idea either, but it might happen), it will cause a bug. In general it might be best to avoid optional parameters in abstract and virtual methods because it's easy for things to go wrong. I would make it a normal parameter.

Reply:
One more question that it didn't occur to me earlier: this method has been overriden only for the ImportSelectNetworkPage, however we have one more page, CrossPoolMigrateNetworkingPage that derives from SelectMultipleVMNetworkPage; do we need to filter SR-IOV there?

Reply:
Thanks. For the CrossPoolMigrateNetworkingPage, we don't need to filter SR-IOV because of the requirement.

Reply:
I'm afraid this is not clear in the req, indeed @MihaelaStoica posed the question in ticket CA-286129, but I don't see an explicit answer. In any case, and due to the time zone difference, we had a quick chat with @robhoes and it should be filtered out. Which means the check can go in the base class. In the one case, whether sriov is allowed will be taken from the VM, in the other from the OVF. Try seeing if it's better as a design to add this in the VmMapping class or make a method in the SelectMultipleVMNetworkPage that the child classes will override.

Reply:
Thanks. I think I have a wrong memory. I also checked the CA-286129 and found out that there was some problems that time. However, actually, I can't remember what is the problem. I think you're right and we need to filter it out.